moonstone/analysis/statistical_test.py

```diff
@@ -0,0 +1,167 @@
+import numpy as np
+import pandas as pd
+from scipy import stats
+from statsmodels.stats.multitest import multipletests
+from typing import List, Union
+
+from moonstone.utils.pandas.series import SeriesBinning
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def statistical_test_groups_comparison(series: pd.Series, group_series: pd.Series, stat_test: str,
+                                       output: str = 'dataframe', mirror: str = True, **kwargs):
+    """
+    :param output: {'series', 'dataframe'}
+
+    In kwargs, you can pass argument for statistical test, like :
+    :param equal_var: For ttest_ind, set to True if your samples have the same variance and
+    you wish to perform a Student's t-test rather than a Welch's t-test (here default is False)
+    :param alternative: {None, 'two-sided', 'less', 'greater'} For Mann - Whitney-U, you can define
+    the alternative hypothesis.
+    :param bins: For chi2_contingency, you can define bins.
+    """
+
+
+    possibles = globals().copy()
+    possibles.update(locals())
+    method = possibles.get(stat_test)
+    if not method:
+        raise NotImplementedError("Method %s not implemented" % stat_test)
+
+    groups = list(group_series.unique())
+    groups.sort()
+
+    list_of_series = []
+    new_groups = []
+    for i in groups:
+        groupi = series[group_series[group_series == i].index].dropna()
+        if groupi.size < 20 and stat_test == 'mann_whitney_u':
+            logger.warning(f'Less than 20 observations for group {i} in data.')
+            list_of_series += [groupi]
+            new_groups += [i]
+        elif groupi.size < 10 and stat_test == 'chi2_contingency':
+            logger.warning(f'Not enough observations (<10) for group {i} in data. Group dropped.')
+        else:
+            list_of_series += [groupi]
+            new_groups += [i]
+
+    if len(new_groups) == 0:
+        raise RuntimeError("All groups have been dropped.")
+
+    if stat_test == 'chi2_contingency' and 'bins' not in kwargs.keys():
+        kwargs['bins'] = _compute_best_bins_values(list_of_series)
+
+    groups = new_groups
+
+    if output == 'series':
+        dic_df = {}
+        for i in range(len(groups)):
+            for j in range(i+1, len(groups)):
+                pval = method(list_of_series[i], list_of_series[j], **kwargs)[1]
+                dic_df[(groups[i], groups[j])] = pval
+                if mirror:
+                    dic_df[(groups[j], groups[i])] = pval
+        mann_whitney_u_df = pd.Series(dic_df)
```

Comment:
here the name of the df is specific to mann whitney, i guess it should be changed?